spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/schemasqlgeneration/SchemaSQLGenerationDataModel.java

```diff
@@ -56,7 +62,9 @@ public SchemaSQLGenerationDataModel(RelationalMappingContext context) {
 
             while (iter.hasNext()) {
                 BasicRelationalPersistentProperty p = iter.next();
-                ColumnModel columnModel = new ColumnModel(p.getColumnName(), p.getActualType());
+                ColumnModel columnModel = new ColumnModel(p.getColumnName(),
+                        typeMapper.databaseTypeFromClass(p.getActualType()),
```

Comment:
@schauder - I know initially I had wanted to modify the @Column annotation to have a gazillion modifiers :-) to match what was in JPA.

After implementing the default behavior (and the change to the Column model) - I think adding just one annotation (databaseType) would be sufficient.     We would let developers specify whatever they want there, and that is exactly what would be in the generated SQL.

If the user wishes to change our defaults, they can optionally subclass BaseTypeMapper.   I stayed with what I think are datatypes that are supported by all Databases.  

Will add more comments to the code - to document/describe to users how to extend/change behavior.